scripts/make_docs_locally.sh

```diff
@@ -4,11 +4,18 @@
 # python scripts/json_schema_to_md_doc.py
 # python scripts/generate_dialect_comparison_docs.py
 
-deactivate
-python3 -m venv docs-venv
-source docs-venv/bin/activate
-pip install --upgrade pip
-pip install -r scripts/docs-requirements.txt
+if [[ "$VIRTUAL_ENV" != "$cwd/docs-venv" ]]; then
+    if [ -n "$VIRTUAL_ENV" ]; then
+        deactivate
+    else    
+        :
+    fi
+    python3 -m venv docs-venv
+    source docs-venv/bin/activate
+    pip install --upgrade pip
+    pip install -r scripts/docs-requirements.txt
+fi
+
 # if you haven't run generate_dialect_comparison_docs.py, then install the splink dependencies and run
 if [[ ! -f "docs/includes/generated_files/comparison_level_library_dialect_table.md" ]]
```

Comment:
We probably also want to make sure this is only run once

Reply:
I might be misunderstanding, but I think it is only run once. The first time the bash script is called, `scripts/generate_dialect_comparison_docs.py` generates `comparison_level_library_dialect_table.md` and so the next time it's called, the file exists and the if statement isn't triggered

Reply:
I'll check on Monday, but I think that block gets run every time because the file isn't generated. Nothing to do with your fix, more an issue with the original implementation.

Reply:
I've checked this and it is an issue w/ how these tables are currently being generated. I'll add another PR in to fix this behaviour.